lib/encode.ex

```diff
@@ -204,6 +205,46 @@ defmodule Jason.Encode do
     end
   end
 
+  defp map_camel_case(value, escape, encode_map) do
+    case Map.to_list(value) do
+      [] -> "{}"
+      [{key, value} | tail] ->
+        key = camelize(key)
+        ["{\"", key(key, escape), "\":",
+         value(value, escape, encode_map)
+         | map_camel_case_loop(tail, escape, encode_map)]
+    end
+  end
+
+  defp map_camel_case_loop([], _escape, _encode_map) do
```

Comment:
```suggestion
  defp do_map_camel_case([], _escape, _encode_map) do
```

Reply:
The reason why I named the functions as above is that on each encode map function there is a pattern in the naming that goes like encode_option and encode_option_loop.